### ISSUE TITLE
Windows: pressing `<C-c>` after `lua os.execute("")` kills neovim

### ISSUE THREAD

**Issue by nazo6** [neovim/neovim#17762]
### Neovim version (nvim -v)

0.7.0-dev+1275-g00effff56

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

Windows11

### Terminal name/version

Windows Terminal 1.11

### $TERM environment variable

vtpcon

### Installation

scoop

### How to reproduce the issue

```
nvim
:lua os.execute("")
````
then press `<C-c>`

This did not happen with Arch Linux on wsl.

### Expected behavior

Neovim is not killed.

### Actual behavior

Neovim is killed and terminal is broken like this.

![image](https://user-images.githubusercontent.com/53177992/159005175-7d967d89-4c67-4746-b6f4-952187e080c1.png)

**Comment by loichyan** [comment]
This also does not happen when using `nvim-qt` or other GUIs like `neovide`.

**Comment by 3N4N** [comment]
Huh! nvim's crashing, right? Funny. I'll dig around a little tomorrow and report back if I find anything.

**Comment by justinmk** [comment]
Not crashing, but appears to be interpreting ctrl-c as SIGINT, possibly because `lua os.execute("")` does something with stdin.

**Comment by 3N4N** [comment]
Right. And it doesn't matter what command, empty or wrong or correct, you pass to `os.execute()`, it will kill nvim on the next Ctrl-C. You can use nvim as usual, as long as you don't press Ctrl-C. It seems nvim's ability to handle Ctrl-C gets busted after `:lua os.execute("os_cmd")` As you said, something to do with stdin.

**Comment by erw7** [comment]
This was caused by a change in the mode of the console. However, #18375 has shown that this problem no longer occurs on servers running in a separate process from `tui`, since `stdin` is not a console. 

I tried using `ffi` and found that it was still possible to get a handle on `CONIN$` with `CreateFile` and change the mode of the console. It is better to call [`FreeConsole`](https://learn.microsoft.com/en-us/windows/console/freeconsole) on the server to detaches the console.
